### ISSUE TITLE
Cannot perform backprop: Dropout mask array is absent (already cleared?) in beta2

### ISSUE THREAD

**Issue by fmorbini** [deeplearning4j/deeplearning4j#6326]
i'm training the following network:

      ComputationGraphConfiguration conf = new NeuralNetConfiguration.Builder()
          .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
          .seed(1000)
          .gradientNormalization(GradientNormalization.RenormalizeL2PerParamType)
          .l2(1e-5)
          .dropOut(0.8)
          .updater(new Adam(3))
          .weightInit(WeightInit.XAVIER).graphBuilder().addInputs("vectors", "ontology")
          .addVertex("merge", new MergeVertex(), "vectors", "ontology")
          .addLayer(LSTM_LAYER,
              new Bidirectional(Bidirectional.Mode.CONCAT, new LSTM.Builder()
                  .nIn(getW2vService().getVectorSize() + ontologySize).nOut(hiddenaLayerSize)
                  .activation(Activation.TANH)
                  .dropOut(new GaussianNoise(0.05))
                  .build())
              ,"merge")
          .addLayer("intentOut",
              new RnnOutputLayer.Builder().activation(Activation.SOFTMAX)
                  .lossFunction(LossFunctions.LossFunction.MCXENT).nIn(hiddenaLayerSize*2)
                  .nOut(intentsDictionary.size()).build(),
              LSTM_LAYER)
          .addLayer("neOut",
              new RnnOutputLayer.Builder().activation(Activation.SOFTMAX)
                  .lossFunction(LossFunctions.LossFunction.MCXENT).nIn(hiddenaLayerSize*2)
                  .nOut(nesDictionary.size()).build(),
              LSTM_LAYER)
          .setOutputs("intentOut", "neOut").build();

on 1.0-alpha and it works fine.

I gave a try to 1.0-beta2 and i get the following exception as soon as training starts:

Exception in thread "main" java.lang.IllegalStateException: Cannot perform backprop: Dropout mask array is absent (already cleared?)
	at org.nd4j.base.Preconditions.throwStateEx(Preconditions.java:626)
	at org.nd4j.base.Preconditions.checkState(Preconditions.java:253)
	at org.deeplearning4j.nn.conf.dropout.Dropout.backprop(Dropout.java:154)
	at org.deeplearning4j.nn.layers.AbstractLayer.backpropDropOutIfPresent(AbstractLayer.java:309)
	at org.deeplearning4j.nn.layers.recurrent.RnnOutputLayer.backpropGradient(RnnOutputLayer.java:70)
	at org.deeplearning4j.nn.graph.vertex.impl.LayerVertex.doBackward(LayerVertex.java:148)
	at org.deeplearning4j.nn.graph.ComputationGraph.calcBackpropGradients(ComputationGraph.java:2592)
	at org.deeplearning4j.nn.graph.ComputationGraph.computeGradientAndScore(ComputationGraph.java:1366)
	at org.deeplearning4j.nn.graph.ComputationGraph.computeGradientAndScore(ComputationGraph.java:1326)
	at org.deeplearning4j.optimize.solvers.BaseOptimizer.gradientAndScore(BaseOptimizer.java:160)
	at org.deeplearning4j.optimize.solvers.StochasticGradientDescent.optimize(StochasticGradientDescent.java:51)
	at org.deeplearning4j.optimize.Solver.optimize(Solver.java:52)
	at org.deeplearning4j.nn.graph.ComputationGraph.fitHelper(ComputationGraph.java:1149)
	at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:1098)
	at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:1065)

**Comment by AlexDBlack** [comment]
I'll need more than just your network configuration - I'll need to see how you are training this too.

**Comment by fmorbini** [comment]
@AlexDBlack here is a gist: https://gist.github.com/fmorbini/936a7f1c9b0aa905e3ecf71dc096cb30
that repros the issue at least on my machine.

it's not the same issue but related to https://github.com/deeplearning4j/deeplearning4j/issues/5571
Unfortunately i have little time to play around with new versions. It takes already some time to modify the code to adhere to the changes in api. Sorry for not following up to all your messages.

thank you

**Comment by AlexDBlack** [comment]
@fmorbini thanks, that's helpful. I'll post here once I've had a chance to run it.

**Comment by AlexDBlack** [comment]
Thanks for reporting - fixed here: https://github.com/deeplearning4j/deeplearning4j/pull/6370
That should be merged in the next 24-hours or so.

FYI the issue is specific to dropout on output layers in computation graph.
You can work around it by removing the dropout from the output layer, or using an explicit dropout layer instead of dropout on the output layer.
